File: PoorPigs2.py
class Solution:
    def poorPigs(self, buckets: int, minutesToDie: int, minutesToTest: int) -> int:
        import math
        return math.ceil(math.log(buckets, minutesToTest // minutesToDie + 1))


if __name__ == '__main__':
    buckets = int(input())
    minutesToDie = int(input())
    minutesToTest = int(input())
    solution = Solution()
    print(solution.poorPigs(buckets, minutesToDie, minutesToTest))

File: PoorPigs2.cpp
#include<iostream>
#include<vector>
#include<math.h>

using namespace std;

class Solution {
public:
    int poorPigs(int buckets, int minutesToDie, int minutesToTest) {
        
        int pigs = 0;
        while(pow(minutesToTest / minutesToDie + 1, pigs) < buckets) pigs++;
        return pigs;
    }
};

int main() {
	
	int buckets, minutesToDie, minutesToTest;
	cin>>buckets>>minutesToDie>>minutesToTest;
	
	Solution *solution = new Solution();
	cout<<solution->poorPigs(buckets, minutesToDie, minutesToTest);
	
	return 0;
	
}
